src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,32 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
 
 public class AsyncRequestProcessor {
+    private static final int SLEEP_TIME = 200;
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture.supplyAsync(() ->
+                cache.computeIfAbsent(userId, this::getUserData), executor);
+    }
+
+    private UserData getUserData(String userId) {
+        try {
+            TimeUnit.MILLISECONDS.sleep(SLEEP_TIME);
```

Comment:
Using `TimeUnit.MILLISECONDS.sleep` to simulate a delay can be misleading in a production environment. If this is a placeholder for a network call or a database query, consider replacing it with actual I/O operations or, if it's for demonstration purposes, document it clearly.